<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Cassava GS SOPs</title>
  <meta name="description" content="Documentation for genomic selection-related standard operating procedures used for NextGen Cassava Breeding." />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content="Cassava GS SOPs" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Documentation for genomic selection-related standard operating procedures used for NextGen Cassava Breeding." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Cassava GS SOPs" />
  
  <meta name="twitter:description" content="Documentation for genomic selection-related standard operating procedures used for NextGen Cassava Breeding." />
  

<meta name="author" content="Marnin Wolfe" />


<meta name="date" content="2020-02-27" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="prepTP.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />









<link href="libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="libs/pagedtable-1.1/js/pagedtable.js"></script>


<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Cassava GS SOPs</a></li>

<li class="divider"></li>
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Overview</a><ul>
<li class="chapter" data-level="0.1" data-path="index.html"><a href="index.html#workflows-use-cases"><i class="fa fa-check"></i><b>0.1</b> Workflows / Use Cases</a></li>
<li class="chapter" data-level="0.2" data-path="index.html"><a href="index.html#example_data"><i class="fa fa-check"></i><b>0.2</b> Example data</a></li>
<li class="chapter" data-level="0.3" data-path="index.html"><a href="index.html#appendices"><i class="fa fa-check"></i><b>0.3</b> Appendices</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="prepTP.html"><a href="prepTP.html"><i class="fa fa-check"></i><b>1</b> Prepare training data</a><ul>
<li class="chapter" data-level="1.1" data-path="prepTP.html"><a href="prepTP.html#cassavabase-download"><i class="fa fa-check"></i><b>1.1</b> Cassavabase download</a></li>
<li class="chapter" data-level="1.2" data-path="prepTP.html"><a href="prepTP.html#formatting-and-initial-qc"><i class="fa fa-check"></i><b>1.2</b> Formatting and initial QC</a><ul>
<li class="chapter" data-level="1.2.1" data-path="prepTP.html"><a href="prepTP.html#checklist"><i class="fa fa-check"></i><b>1.2.1</b> Checklist</a></li>
<li class="chapter" data-level="1.2.2" data-path="prepTP.html"><a href="prepTP.html#specifications-for-output"><i class="fa fa-check"></i><b>1.2.2</b> Specifications for Output</a></li>
</ul></li>
<li class="chapter" data-level="1.3" data-path="prepTP.html"><a href="prepTP.html#worked-example"><i class="fa fa-check"></i><b>1.3</b> Worked example</a><ul>
<li class="chapter" data-level="1.3.1" data-path="prepTP.html"><a href="prepTP.html#read-in-data"><i class="fa fa-check"></i><b>1.3.1</b> Read-in data</a></li>
<li class="chapter" data-level="1.3.2" data-path="prepTP.html"><a href="prepTP.html#traits-and-traitabbreviations"><i class="fa fa-check"></i><b>1.3.2</b> Traits and TraitAbbreviations</a></li>
<li class="chapter" data-level="1.3.3" data-path="prepTP.html"><a href="prepTP.html#qc-trait-values"><i class="fa fa-check"></i><b>1.3.3</b> QC Trait values</a></li>
<li class="chapter" data-level="1.3.4" data-path="prepTP.html"><a href="prepTP.html#harvest-index"><i class="fa fa-check"></i><b>1.3.4</b> Harvest Index</a></li>
<li class="chapter" data-level="1.3.5" data-path="prepTP.html"><a href="prepTP.html#unit-area-traits"><i class="fa fa-check"></i><b>1.3.5</b> Unit area traits</a></li>
<li class="chapter" data-level="1.3.6" data-path="prepTP.html"><a href="prepTP.html#season-wide-mean-cmds"><i class="fa fa-check"></i><b>1.3.6</b> Season-wide mean CMDS</a></li>
<li class="chapter" data-level="1.3.7" data-path="prepTP.html"><a href="prepTP.html#assign-genos-to-phenos"><i class="fa fa-check"></i><b>1.3.7</b> Assign genos to phenos</a></li>
<li class="chapter" data-level="1.3.8" data-path="prepTP.html"><a href="prepTP.html#write-cleaned-dataset"><i class="fa fa-check"></i><b>1.3.8</b> Write cleaned dataset</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="2" data-path="curateTrials.html"><a href="curateTrials.html"><i class="fa fa-check"></i><b>2</b> Curate by trial</a><ul>
<li class="chapter" data-level="2.1" data-path="curateTrials.html"><a href="curateTrials.html#worked-example-1"><i class="fa fa-check"></i><b>2.1</b> Worked example</a><ul>
<li class="chapter" data-level="2.1.1" data-path="curateTrials.html"><a href="curateTrials.html#format-input-data"><i class="fa fa-check"></i><b>2.1.1</b> Format input data</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="curateTrials.html"><a href="curateTrials.html#under-construction"><i class="fa fa-check"></i><b>2.2</b> UNDER CONSTRUCTION</a><ul>
<li class="chapter" data-level="2.2.1" data-path="curateTrials.html"><a href="curateTrials.html#function-to-analyze-trials"><i class="fa fa-check"></i><b>2.2.1</b> Function to analyze trials</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="3" data-path="getBLUPs.html"><a href="getBLUPs.html"><i class="fa fa-check"></i><b>3</b> Get BLUPs</a><ul>
<li class="chapter" data-level="3.1" data-path="getBLUPs.html"><a href="getBLUPs.html#explanation-of-two-stage-prediction"><i class="fa fa-check"></i><b>3.1</b> Explanation of two-stage prediction</a></li>
<li class="chapter" data-level="3.2" data-path="getBLUPs.html"><a href="getBLUPs.html#set-up-training-data-chunks"><i class="fa fa-check"></i><b>3.2</b> Set-up training data chunks</a></li>
</ul></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Cassava GS SOPs</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">Cassava GS SOPs</h1>
<p class="author"><em>Marnin Wolfe</em></p>
<p class="date"><em>2020-02-27</em></p>
</div>
<div id="overview" class="section level1 unnumbered">
<h1>Overview</h1>
<div id="workflows-use-cases" class="section level2">
<h2><span class="header-section-number">0.1</span> Workflows / Use Cases</h2>
<p>The purpose of this document is as a manual, a teaching tool and a codebase to support genomic selection-related standard operating procedures (SOPs). This should serve to support the <a href="http://www.nextgencassava.org">NextGen Cassava Breeding Project</a> and its community of practice partnerships.</p>
<p>Initially, I will attempt to outline the core steps, providing standardized code where possible, and clearly pointing out the areas where breeding programs / analysts will need to provide input.</p>
<p>The actual GS analyses that I conducted in 2019 are documented on GitHub for <a href="https://wolfemd.github.io/IITA_2019GS/">IITA</a>, <a href="https://wolfemd.github.io/NRCRI_2019GS/">NRCRI</a> and <a href="https://wolfemd.github.io/NaCRRI_2019GS/">NaCRRI</a>. Some of the datasets involved are too big for GitHub; the most important ones can be found in <a href="https://drive.google.com/drive/folders/1sjFFS4sRAwxHALEcT8RTSuUdHJsIuGNK?usp=sharing">this</a> Google Drive folder.</p>
<p><strong>Use Case 1: </strong> Selection of parents for crossing.<br />
Analysis: Using GEBV (standard GBLUP model)</p>
<ol style="list-style-type: decimal">
<li>Step <a href="prepTP.html#prepTP">1</a>: Prepare a training dataset</li>
</ol>
<ul>
<li>Download data from DB, “Clean” and format DB data</li>
</ul>
<ol start="2" style="list-style-type: decimal">
<li>Step <a href="curateTrials.html#curateTrials">2</a>: Curate trials</li>
</ol>
<ul>
<li>remove outliers, select best model for each trial</li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Step <a href="getBLUPs.html#getBLUPs">3</a>: Getting BLUPs from multiple trials</li>
<li>Step <a href="#getKinship"><strong>??</strong></a>: Review and QC of SNP marker data</li>
</ol>
<ul>
<li>construction kinship matrix, PCA to check pop. structure</li>
</ul>
<ol start="5" style="list-style-type: decimal">
<li>Step <a href="#crossValidation"><strong>??</strong></a>: Evaluate prediction accuracy with cross-validation</li>
<li>Step <a href="#getGEBVs"><strong>??</strong></a>: Genomic prediction, get GEBVs for all selection candidates</li>
<li>Step <a href="#getGainEst"><strong>??</strong></a>: Genomic selection</li>
</ol>
<ul>
<li>Estimate genetic gain, Formulate selection index, upload GEBV to DB</li>
</ul>
<p><strong>Use Case 2: </strong> Select clones for further phenotyping<br />
<strong>Use Case 2a: </strong> Advancement of clones for variety development (e.g. harvest current PYTs and advance to AYTs)<br />
Analysis: Using GETGV (non-additive genomic prediction models)<br />
<strong>Use Case 2b: </strong> Maintenance of diversity and prediction accuracy<br />
Analysis: Using STPGA (R package for selecting “optimal” subsets for phenotyping)</p>
</div>
<div id="example_data" class="section level2">
<h2><span class="header-section-number">0.2</span> Example data</h2>
<p>Want to develop some standard functions using a small (50Mb or less dataset). Will start with some previously processed data and use it to select a subset of raw trials to use as examples.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1"></a><span class="kw">library</span>(tidyverse); <span class="kw">library</span>(magrittr)</span>
<span id="cb1-2"><a href="index.html#cb1-2"></a>cleanphenos &lt;-<span class="st"> </span><span class="kw">readRDS</span>(<span class="kw">url</span>(<span class="st">&quot;https://raw.github.com/wolfemd/IITA_2019GS/master/data/IITA_ExptDesignsDetected_72619.rds&quot;</span>))</span>
<span id="cb1-3"><a href="index.html#cb1-3"></a>cleanphenos2keep&lt;-cleanphenos <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb1-4"><a href="index.html#cb1-4"></a><span class="st">      </span><span class="kw">filter</span>(<span class="kw">grepl</span>(<span class="st">&quot;.GS.|gain&quot;</span>,studyName,<span class="dt">ignore.case =</span> T),</span>
<span id="cb1-5"><a href="index.html#cb1-5"></a>             studyYear <span class="op">%in%</span><span class="st"> </span><span class="kw">c</span>(<span class="st">&quot;2016&quot;</span>,<span class="st">&quot;2017&quot;</span>),</span>
<span id="cb1-6"><a href="index.html#cb1-6"></a>             <span class="op">!</span><span class="kw">grepl</span>(<span class="st">&quot;ungenotyped&quot;</span>,studyName))</span>
<span id="cb1-7"><a href="index.html#cb1-7"></a>trialsForExample&lt;-<span class="kw">unique</span>(cleanphenos2keep<span class="op">$</span>studyName)</span></code></pre></div>
<p>I’ve chosen 2 years of IITA data, just the GS-related trials.</p>
<p>Should encompass RCBD and Augmented designs and the range of plot sizes.</p>
<p>Next part will call on data located <a href="https://drive.google.com/drive/folders/1sjFFS4sRAwxHALEcT8RTSuUdHJsIuGNK?usp=sharing">here</a>, subset the raw data and save as example data.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1"></a>rawphenos&lt;-<span class="kw">read.csv</span>(<span class="st">&quot;~/Google Drive/NextGenGS/NGC_BigData/DatabaseDownload_72419/2019-07-24T144915phenotype_download.csv&quot;</span>,</span>
<span id="cb2-2"><a href="index.html#cb2-2"></a>         <span class="dt">na.strings =</span> <span class="kw">c</span>(<span class="st">&quot;#VALUE!&quot;</span>,<span class="ot">NA</span>,<span class="st">&quot;.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="st">&quot; &quot;</span>,<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>),</span>
<span id="cb2-3"><a href="index.html#cb2-3"></a>         <span class="dt">stringsAsFactors =</span> F)</span>
<span id="cb2-4"><a href="index.html#cb2-4"></a>examplerawphenos&lt;-rawphenos <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-5"><a href="index.html#cb2-5"></a><span class="st">      </span><span class="kw">filter</span>(studyName <span class="op">%in%</span><span class="st"> </span>trialsForExample)</span>
<span id="cb2-6"><a href="index.html#cb2-6"></a><span class="kw">write.csv</span>(examplerawphenos,<span class="dt">file=</span><span class="st">&quot;data/example_phenotype_download.csv&quot;</span>,<span class="dt">row.names =</span> F)</span>
<span id="cb2-7"><a href="index.html#cb2-7"></a></span>
<span id="cb2-8"><a href="index.html#cb2-8"></a><span class="kw">read.csv</span>(<span class="st">&quot;~/Google Drive/NextGenGS/NGC_BigData/DatabaseDownload_72419/2019-07-24T144144metadata_download.csv&quot;</span>,</span>
<span id="cb2-9"><a href="index.html#cb2-9"></a>         <span class="dt">na.strings =</span> <span class="kw">c</span>(<span class="st">&quot;#VALUE!&quot;</span>,<span class="ot">NA</span>,<span class="st">&quot;.&quot;</span>,<span class="st">&quot;&quot;</span>,<span class="st">&quot; &quot;</span>,<span class="st">&quot;-&quot;</span>,<span class="st">&quot;</span><span class="ch">\&quot;</span><span class="st">&quot;</span>),</span>
<span id="cb2-10"><a href="index.html#cb2-10"></a>         <span class="dt">stringsAsFactors =</span> F) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-11"><a href="index.html#cb2-11"></a><span class="st">      </span><span class="kw">filter</span>(studyName <span class="op">%in%</span><span class="st"> </span>trialsForExample) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb2-12"><a href="index.html#cb2-12"></a><span class="st">      </span><span class="kw">write.csv</span>(.,<span class="dt">file=</span><span class="st">&quot;data/example_metadata_download.csv&quot;</span>,<span class="dt">row.names =</span> F)</span></code></pre></div>
<p>Let’s also make an example SNP dataset. Will sample a subset of SNPs and remove unecessary samples to make an example dosage matrix.</p>
<p>Sample 2000 SNPs. Create an easy to use data.frame <strong>geno2pheno</strong> with 2 columns:</p>
<ul>
<li><strong>germplasmName</strong> matches the cassavabase phenotype data download.</li>
<li><strong>DNAname:</strong> matches the <code>rownames</code> of the example snps dosage matrix <strong>snps</strong>.</li>
</ul>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1"></a>snps&lt;-<span class="kw">readRDS</span>(<span class="st">&quot;~/Google Drive/NextGenGS/NGC_BigData/DosageMatrix_RefPanelAndGSprogeny_ReadyForGP_73019.rds&quot;</span>)</span>
<span id="cb3-2"><a href="index.html#cb3-2"></a><span class="kw">dim</span>(snps) <span class="co"># [1] 21856 68814</span></span>
<span id="cb3-3"><a href="index.html#cb3-3"></a></span>
<span id="cb3-4"><a href="index.html#cb3-4"></a>cleanphenos2keep&lt;-cleanphenos2keep <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-5"><a href="index.html#cb3-5"></a><span class="st">      </span><span class="kw">select</span>(studyName,TrialData) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-6"><a href="index.html#cb3-6"></a><span class="st">      </span><span class="kw">unnest</span>(<span class="dt">cols =</span> <span class="kw">c</span>(TrialData))</span>
<span id="cb3-7"><a href="index.html#cb3-7"></a></span>
<span id="cb3-8"><a href="index.html#cb3-8"></a>geno2pheno&lt;-cleanphenos2keep <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-9"><a href="index.html#cb3-9"></a><span class="st">      </span><span class="kw">select</span>(germplasmName,FullSampleName) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-10"><a href="index.html#cb3-10"></a><span class="st">      </span>distinct <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-11"><a href="index.html#cb3-11"></a><span class="st">      </span><span class="kw">filter</span>(<span class="op">!</span><span class="kw">is.na</span>(FullSampleName),</span>
<span id="cb3-12"><a href="index.html#cb3-12"></a>             FullSampleName <span class="op">%in%</span><span class="st"> </span><span class="kw">rownames</span>(snps)) <span class="op">%&gt;%</span><span class="st"> </span></span>
<span id="cb3-13"><a href="index.html#cb3-13"></a><span class="st">      </span><span class="kw">rename</span>(<span class="dt">DNAname=</span>FullSampleName)</span>
<span id="cb3-14"><a href="index.html#cb3-14"></a></span>
<span id="cb3-15"><a href="index.html#cb3-15"></a><span class="kw">set.seed</span>(<span class="dv">42</span>)</span>
<span id="cb3-16"><a href="index.html#cb3-16"></a>snps&lt;-snps[<span class="kw">rownames</span>(snps) <span class="op">%in%</span><span class="st"> </span>geno2pheno<span class="op">$</span>DNAname,<span class="kw">sample</span>(<span class="dv">1</span><span class="op">:</span><span class="kw">ncol</span>(snps),<span class="dt">size =</span> <span class="dv">2000</span>, <span class="dt">replace =</span> F)]</span>
<span id="cb3-17"><a href="index.html#cb3-17"></a></span>
<span id="cb3-18"><a href="index.html#cb3-18"></a><span class="kw">saveRDS</span>(snps,<span class="dt">file=</span><span class="st">&quot;data/example_snps.rds&quot;</span>)</span>
<span id="cb3-19"><a href="index.html#cb3-19"></a><span class="kw">saveRDS</span>(geno2pheno,<span class="dt">file=</span><span class="st">&quot;data/example_geno2pheno.rds&quot;</span>)</span></code></pre></div>
</div>
<div id="appendices" class="section level2">
<h2><span class="header-section-number">0.3</span> Appendices</h2>
<ol style="list-style-type: decimal">
<li>Access to and use of computing resources</li>
<li>Learning resources</li>
</ol>
<ul>
<li><a href="https://youtu.be/Qne86lxjgtg" class="uri">https://youtu.be/Qne86lxjgtg</a></li>
<li><a href="https://youtu.be/bzUmK0Y07ck" class="uri">https://youtu.be/bzUmK0Y07ck</a></li>
<li><a href="https://youtu.be/D48JHU4llkk" class="uri">https://youtu.be/D48JHU4llkk</a></li>
<li><a href="https://r4ds.had.co.nz/">R for Data Science</a></li>
</ul>
<ol start="3" style="list-style-type: decimal">
<li>Tidy data</li>
</ol>
<ul>
<li><ol style="list-style-type: decimal">
<li>each variable in a column, (2) each observation in a row, (3) each cell is one value</li>
</ol></li>
</ul>
<ol start="4" style="list-style-type: decimal">
<li>Misc</li>
</ol>
<ul>
<li><a href="https://malco.io/2018/11/05/why-should-i-use-the-here-package-when-i-m-already-using-projects/" class="uri">https://malco.io/2018/11/05/why-should-i-use-the-here-package-when-i-m-already-using-projects/</a></li>
<li>[Wilson et al. 2017. <em>PloS Computation Biology</em>.] Good enough practices in scientific computing. (<a href="https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510" class="uri">https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510</a>)</li>
</ul>

</div>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="prepTP.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["CassavaGS_SOPs.pdf", "CassavaGS_SOPs.epub"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
